src/libAtomVM/refc_binary.c

```diff
@@ -55,16 +55,23 @@ void refc_binary_increment_refcount(struct RefcBinary *refc)
     refc->ref_count++;
 }
 
-bool refc_binary_decrement_refcount(struct RefcBinary *refc)
+bool refc_binary_decrement_refcount(struct RefcBinary *refc, struct GlobalContext *global)
 {
     if (--refc->ref_count == 0) {
-        list_remove(&refc->head);
-        free(refc);
+        synclist_remove(&global->refc_binaries, &refc->head);
+        refc_binary_destroy(refc, global);
         return true;
     }
     return false;
 }
 
+void refc_binary_destroy(struct RefcBinary *refc, struct GlobalContext *global)
```

Comment:
Are you passing in the GlobalContext for compatibility with resources, for example, when a resource d'tor needs it?

Reply:
Yes it's paving the way towards passing an environment to the dtor, even if it's NULL in 599 for now. We'll see if you need one for sockets :-)